core-implementation/src/main/java/gg/solarmc/loader/impl/player/DelegatingSolarPlayer.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * DataLoader
+ * Copyright © 2021 SolarMC Developers
+ *
+ * DataLoader is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * DataLoader is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with DataLoader. If not, see <https://www.gnu.org/licenses/>
+ * and navigate to version 3 of the GNU Affero General Public License.
+ */
+
+package gg.solarmc.loader.impl.player;
+
+import gg.solarmc.loader.OnlineSolarPlayer;
+import gg.solarmc.loader.SolarPlayer;
+import gg.solarmc.loader.data.DataKey;
+import gg.solarmc.loader.data.DataObject;
+import gg.solarmc.loader.impl.PlayerTracker;
+
+import java.util.Optional;
+import java.util.UUID;
+
+public final class DelegatingSolarPlayer implements SolarPlayer {
```

Comment:
This here is an "offline player"?

Reply:
Most of the time. If the user is actually online, DelegatingSolarPlayer delegates to the online SolarPlayer. This means that, if you acquire an offline solar player, then an online solar player logs in, the offline player will delegate to the data of the online player.

Otherwise, DelegatingSolarPlayer uses its own data objects.